### PR TITLE
Make sure RAT plugin ignores files in the target folder

### PR DESCRIPTION
**Pull request by gnodet**

### CODE REVIEW

**Comment by cstamas** [comment]
better but not enough... if you do `-Dmaven.repo.local=local` RAT will belly up AGAIN.... best would be to make it smarter, but.. sigh.

**Review by cstamas — APPROVED** [review]

**Comment by michael-o** [comment]
> better but not enough... if you do `-Dmaven.repo.local=local` RAT will belly up AGAIN.... best would be to make it smarter, but.. sigh.


Well, you can't make thid right. I basically do `...=$PWD/../repo` or `...=$(mktemp -d)`

File: pom.xml [hunk 1]
```diff
@@ -599,6 +599,8 @@ under the License.
               <exclude>src/main/appended-resources/licenses/EPL-1.0.txt</exclude>
               <exclude>src/main/appended-resources/licenses/unrecognized-aopalliance-1.0.txt</exclude>
               <exclude>src/main/appended-resources/licenses/unrecognized-javax.annotation-api-1.3.2.txt</exclude>
+              <!-- exclude all files in the target folder -->
```
Comment (michael-o) [inline pom.xml:602]: directory, not folder

**Review by michael-o — CHANGES_REQUESTED** [review]

**Comment by cstamas** [comment]
> Well, you can't make thid right. I basically do `...=$PWD/../repo` or `...=$(mktemp -d)`

So RAT scanns ALL files from project basedir? How it avoid scanning files multiple times, like when project modules are nested? Are they being over and over re-scanned as build progresses (and RAT plugin is inherited from top level module)?

**Comment by michael-o** [comment]
> > Well, you can't make this right. I basically do `...=$PWD/../repo` or `...=$(mktemp -d)`
> 
> So RAT scanns ALL files from project basedir? How it avoid scanning files multiple times, like when project modules are nested? Are they being over and over re-scanned as build progresses (and RAT plugin is inherited from top level module)?

For this we need to check the source code of this plugin.

**Pull request merged**